src/binding/petsc4py/src/petsc4py/PETSc/Mat.pyx

```diff
@@ -455,7 +836,32 @@ cdef class Mat(Object):
 
     #
 
-    def createDense(self, size, bsize=None, array=None, comm=None):
+    def createDense(
+        self,
+        size: MatSizeType,
+        bsize: MatBlockSizeType | None = None,
+        array: Sequence[Scalar] | None = None,
+        comm: Comm | None = None
+    ) -> Self:
+        """Create a dense matrix.
+
+        Collective.
+
+        Parameters
+        ----------
+        size, bsize
+            Matrix size parameters. See `Mat.createAIJ` for usage information.
+        array
+            Matrix data. If `None` then the matrix will not be preallocated.
```

Comment:
This is wrong I guess. If None, the data is internally allocated by petsc, while if it is nonzero it uses userdefined memory. Please double check